lib/ansible/galaxy/api.py

```diff
@@ -63,25 +66,32 @@ class GalaxyAPI(object):
 
     SUPPORTED_VERSIONS = ['v1']
 
-    def __init__(self, galaxy, name, url, username=None, password=None, token=None):
+    def __init__(self, galaxy, name, url, username=None, password=None, token=None, token_type=None):
         self.galaxy = galaxy
         self.name = name
         self.username = username
         self.password = password
         self.token = token
+        self.token_type = token_type or 'Token'
         self.api_server = url
         self.validate_certs = not context.CLIARGS['ignore_certs']
         self.baseurl = None
         self.version = None
         self.initialized = False
+        self.available_api_versions = {}
```

Comment:
Leading into my other comments, I think this should be lazily evaluated on this class when called for the first time to save the time it takes for `ansible-galaxy` to complete an operation.